test/no-array-instanceof.js

```diff
@@ -16,13 +18,11 @@ const errors = [{
 ruleTester.run('avoid-array-instanceof', rule, {
 	valid: [
 		'Array.isArray(arr)',
-		'Array.isArray(arr) === true',
-		'Array.isArray([])',
-		'Array.isArray([]) === true',
-		'Array.isArray([1,2,3])',
-		'Array.isArray([1,2,3]) === true',
-		'if(Array.isArray(arr)){}',
-		'if(Array.isArray([1,2,3])){}'
+		'arr instanceof Object',
+		'arr instanceof array',
+		'a instanceof \'array\'',
+		'a instanceof ArrayA',
+		`a.x[${value}] instanceof Object`
```

Comment:
You should have a test where the right side of the operator is not an identifier (ok, you already have strings, but another test can't hurt). Maybe `arr instanceof foo()`